File: Potion.cpp
//
//  Potion.cpp
//  
//
//  Created by Muhammad Sohibul Maromi on 3/15/13.
//
//

#include "Potion.h"

Potion::Potion(int x, string s, int mp, int hp) : Item(100,"potion") { //ini kok x sama s ngga ada gunanya?
    mpTambah = mp;
    hpTambah = hp;
}

Potion::Potion(int _mp, int _hp) : Item(100,"potion"){
	this->mpTambah = _mp;
	this->hpTambah = _hp;
}

Potion::Potion(const Potion& p) {
    mpTambah = p.mpTambah;
    hpTambah = p.hpTambah;
	
	harga = p.harga;
	itemName = p.itemName;
}

Potion::~Potion() {
    
}

Potion& Potion::operator=(const Potion& p) {
    mpTambah = p.mpTambah;
    hpTambah = p.hpTambah;
	
	harga = p.harga;
	itemName = p.itemName;
    
    return *this;
}

void Potion::setMP(int _mp){
	this->mpTambah = _mp;
}

void Potion::setHP(int _hp){
	this->hpTambah = _hp;
}

void Potion::execute(Monster& m) {
    m.setCurrentHP(m.getCurrentHP() + hpTambah);
    m.setCurrentMP(m.getCurrentMP() + mpTambah);
}

File: Android/model/Potion.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

/**
 *
 * @author msmaromi
 */
public class Potion extends Item {
  private int mpTambah;
  private int hpTambah;
  
  public Potion(int harga, String nama, int mp, int hp) {
    super(harga, nama);
    mpTambah = mp;
    hpTambah = hp;
  }
  
  public Potion(Potion p) {
    super(p.harga, p.nama);
    mpTambah = p.mpTambah;
    hpTambah = p.hpTambah;
  }
  
  public int getMPTambah() {
    return mpTambah;
  }
  
  public int getHPTambah() {
    return hpTambah;    
  }
  
  public void setMPTambah(int mp) {
    mpTambah = mp;
  }
  
  public void setHPTambah(int hp) {
    hpTambah = hp;
  }
  
  @Override
  public void execute(Monster m) {
//    m.setCurrentHP(m.getCurrentHP()+hpTambah);
//    m.setCurrentMP(m.getCurrentMP()+mpTambah);
    
  }
  
}
